Document max_examples cheatcode (#690)
software-mansion/protostar: website/docs/tutorials/07-testing/02-cheatcodes/max-examples.md

# `max_examples`

```python
def max_examples(max_examples: int):
```

> Sets the maximum value of examples to explore by the fuzzer.

Reviewer: ```suggestion
Sets the maximum number of examples to explore by the fuzzer.
```

> Once this many satisfying input cases have been evaluated without finding any counter-example,
> fuzzer falsification will terminate.

Reviewer: I wonder if it wouldn't be simpler to read
```suggestion
Fuzzer tries at most this many input examples. If it won't find any failing, test will pass.
```
Don't need to change though

:::warning
This cheatcode is only available in [setup hooks](../README.md#__setup__).
:::

Protostar defaults to 100 examples.
This default value is chosen to suit a workflow where the test will be part of a suite that
is regularly executed locally or on a CI server,
balancing total running time against the chance of missing a bug.

> If you are writing one-off tests, running tens of thousands of examples is quite reasonable

Reviewer: I think no everyone may understand what one-off tests means.

Author: Done

as Protostar's fuzzer may miss uncommon bugs with default settings.

This cheatcode has no effects on standard test cases.

> ```cairo title="Example"

Reviewer: Can we move example up, above warning?

It is more important and right now it is not clear what it refers to

Author: I moved it just under the warning to match style of assume/reject docs. Good catch!
